### PR TITLE
Inheritance Handling

### PR DESCRIPTION
**Pull request by tvillaren**
Based on #310

### CODE REVIEW

**Comment by kiaking** [comment]
@tvillaren Oh hey never mind, sorry it was my fault. I was so stupid I wasn’t linking to the code that included PR… Sorry for that. Now it’s working. I'll keep testing in my project 👍

**Comment by kiaking** [comment]
@tvillaren OK I've confirmed this is working! 🎉  Do you have any further work to do?

File: src/query/Query.ts [hunk 1]
```diff
@@ -1076,8 +1135,33 @@ export default class Query<T extends Model = Model> {
...
+
+    if(forceModel !== undefined) {
+      return new forceModel(record)
+    }
+
+    let model = this.model 
+
+    // If the record has the right typeKey attribute set, and Model has type mapping 
+    // we hydrate it as the corresponding model
+    if (record && record[model.typeKey] && Object.keys(model.types()).length > 0) {
```
Comment (trevorpe) [inline src/query/Query.ts]: Could this be replaced by the `hasTypes()` method that I noticed above, for reducing code duplication?
```suggestion
    if (record && record[model.typeKey] && model.hasTypes()) {
```

Comment (tvillaren) [reply src/query/Query.ts]: Thanks, just applied the change!

**Comment by tvillaren** [comment]
Hey @kiaking,

Just fixed a bug I had missed (I guess my tests were not covering 100% of use cases).
Anyway, it works again and I included @trevorpe suggestion for removing code repetiton.

**Comment by kiaking** [comment]
@tvillaren Thank you so much! Just need to check why test coverage has dropped, I'll try look into it.

**Comment by kiaking** [comment]
OK now I think this ready, I'll merge it. I've done some huge doc refactor so after merging this one, I'll import this docs to that docs and make a release!

**Pull request merged**